### PR TITLE
Feature/evaluation command

### LINKED ISSUE #157: Update main to evaluate projects with evaluation command
Given a project and its tracedata,  tygegen __init__ main should: 
1. Collect the trace data (Already implemented)
2. Process it using the filters (Already implemented)
3. Generate the files with traced type hints  (Already implemented)
4. Compare the original with the generated file (Classes implemented, but usage missing)
5. Get the metric data (Classes implemented, but usage missing)

Implement step 4 and 5.

### PR DESCRIPTION
**Pull request by RecurvedBow**
Closes #157

### CODE REVIEW

File: typegen/evaluation/__init__.py [hunk 1]
```diff
@@ -6,4 +11,73 @@
...
+    original_typehint_data = file_type_hints_collector.typehint_data
+    file_type_hints_collector.collect_data(traced_path, traced_file_paths_to_compare)
+    traced_typehint_data = file_type_hints_collector.typehint_data
+
+    metric_data = metric_data_calculator.get_metric_data(original_typehint_data, traced_typehint_data)
+    completeness, correctness = metric_data_calculator.get_total_completeness_and_correctness(metric_data)
+    print(f"Completeness: {completeness * 100}%, Correctness: {correctness * 100}%")
+
+
+def _get_changed_file_paths(original_root_folder_path, traced_root_folder_path, potential_changed_files_relative_paths):
```
Comment (bengsparks) [inline typegen/evaluation/__init__.py:72]: Add type hints to this file please.

File: typegen/evaluation/file_type_hints_collector.py [hunk 2]
```diff
@@ -258,12 +277,18 @@ def _get_annotation_value_of_name(self, annotation: cst.Name) -> str:
             current_annotation = type_name
         return self._add_full_module_name_to_annotation(current_annotation)
 
-    def _get_annotation_value_of_attribute(self, annotation: cst.Attribute) -> str:
-        assert isinstance(annotation.value, cst.Name)
-        module_name = annotation.value.value
+    def _get_annotation_value_of_attribute(self, annotation: cst.Attribute, add_full_module_name: bool =True) -> str:
```
Comment (bengsparks) [inline typegen/evaluation/file_type_hints_collector.py:280]: Formatting

**Review by bengsparks — CHANGES_REQUESTED** [review]

File: typegen/evaluation/file_type_hints_collector.py (hunk 2 above)
Comment (RecurvedBow) [reply typegen/evaluation/file_type_hints_collector.py:280]: Fixed

File: typegen/evaluation/__init__.py (hunk 1 above)
Comment (RecurvedBow) [reply typegen/evaluation/__init__.py]: Added

**Review by bengsparks — APPROVED** [review]

**Pull request merged**